backend/src/main/java/com/bakdata/conquery/models/query/concept/SelectResultInfo.java

```diff
@@ -0,0 +1,22 @@
+package com.bakdata.conquery.models.query.concept;
+
+import com.bakdata.conquery.models.concepts.select.Select;
+import com.bakdata.conquery.models.externalservice.ResultType;
+
+import lombok.Getter;
+import lombok.NonNull;
+
+@Getter
+public class SelectResultInfo extends ResultInfo {
+	private final Select select;
+
+	public SelectResultInfo(String name, ResultType type, Integer sameNameOcurrences, int postfix, Select select) {
+		super(name, type, sameNameOcurrences, postfix);
+		this.select = select;
+	}
+	
+	public SelectResultInfo withName(@NonNull String name) {
+	    if (name == null) throw new java.lang.NullPointerException("name");
+		return this.getName() == name ? this : new SelectResultInfo(name, this.getType(), this.getSameNameOcurrences(), this.getPostfix(), this.getSelect());
```

Comment:
This is not nested inside the if, but looks like it, could you clean this up a little?

Reply:
Done. Check was already covered by annotation